### PR TITLE
Removes the extra visibility from access_loggers.stream

### PR DESCRIPTION
**Pull request by mathetake**
Commit Message: Removes the extra visibility from access_loggers.stream
Additional Description:
In the effort to enforce the visibility policy of extensions #35026,
this removes `//test:__subpackages__` visibility from stream access logger
extension. To achieve that, this migrates its configuration tests previously 
placed in test/common/access_log/ to the correct location in test/extensions/**.

Risk Level: none
Testing: done
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

File: test/server/config_validation/server_test.cc [hunk 1]
```diff
@@ -267,12 +267,11 @@ TEST_P(ValidationServerTest, WithProcessContext) {
...
 // exists.)
 
-auto testing_values =
-    ::testing::Values("front-proxy_envoy.yaml", "envoyproxy_io_proxy.yaml",
+auto testing_values = ::testing::Values("front-proxy_envoy.yaml", "envoyproxy_io_proxy.yaml",
 #if defined(WIN32) && defined(SO_ORIGINAL_DST)
-                      "configs_original-dst-cluster_proxy_config.yaml",
+                                        "configs_original-dst-cluster_proxy_config.yaml",
 #endif
-                      "grpc-bridge_server_envoy-proxy.yaml", "front-proxy_service-envoy.yaml");
```
Comment (mathetake) [inline test/server/config_validation/server_test.cc:275]: `grpc-bridge_server_envoy-proxy.yaml` was using extensions inside and was failing. It is almost identical to `front-proxy_service-envoy.yaml` here, so it should be fine to just remove it.

Comment (htuch) [reply test/server/config_validation/server_test.cc:275]: I'm a bit confsued by this comment, does this PR make previously valid configs now stop passing tests?

Comment (mathetake) [reply test/server/config_validation/server_test.cc:275]: not really. This config validation test case (grpc-bridge_server_envoy-proxy.yaml) required the extension compiled in this integration test build target, and that's why it failed if we remove the extension dependency here.

Having said that, I found the fix without requiring test (actually `//test/server:conifg_validation` is allowed to have extensions target in the visibility policy). I will push the commit soon.

Comment (mathetake) [reply test/server/config_validation/server_test.cc]: https://github.com/envoyproxy/envoy/pull/35049/commits/3a8eb40c08ce7ea0d5b0d2d3c6ca9d3bd4a6620b done!

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**